### DISCUSSION TITLE
ENCRYPTION_KEY= #insert a secure key to encrypt

### DISCUSSION THREAD

**Discussion by kierank01** [teslamate-org/teslamate#2483] [Getting started] [answered]
Hi, getting a model 3 next week, can't wait, but setting up teslamate is helping, lol...

Can anyone suggest a good tool to generate a secure key for the tesla api key?

I have looked at a number of tutorials, and some(all I have seen) seem to just delete this line from their config files, without explaining why.

Thanks a lot.

**Comment by adriankumpf** [comment, marked as answer]
The same guidelines apply to the encryption key as to strong passwords: 

- Long passwords are more secure
- Make your password unique

There are no limits on character length or characters allowed.

**Reply by parkr** [reply]
Try a password manager to generate your key. 1Password, LastPass, etc. Once you set it, make sure you don't change it until you're ready to reauthenticate for teslamate.

**Reply by natrlhy** [reply]
With special characters like @&, etc I got this type of error. I tried enclosing it with "" as well
`WARNING: The QYNdPRkr2UTCCE variable is not set. Defaulting to a blank string.`

**Reply by neotaruntius** [reply]
> The same guidelines apply to the encryption key as to strong passwords:
> 
> * Long passwords are more secure
> * Make your password unique
> 
> There are no limits on character length or characters allowed.

So in this case, we're just looking to input a string of characters, just like we do with a password. Right?
